internal/iapl/policy.go

```diff
@@ -289,10 +289,15 @@ func (v *policy) validateResourceTypes() error {
 	for _, resourceType := range v.rt {
 		for _, rel := range resourceType.Relationships {
 			for _, tt := range rel.TargetTypes {
-				if _, ok := v.rt[tt.Name]; !ok {
+				rr, ok := v.rt[tt.Name]
+				if !ok {
 					return fmt.Errorf("%s: relationships: %s: %w", resourceType.Name, tt.Name, ErrorUnknownType)
 				}
 
+				if _, err := gidx.NewID(rr.IDPrefix); err != nil {
+					return fmt.Errorf("%s: relationships: %s: %w", resourceType.Name, tt.Name, err)
+				}
```

Comment:
This seems to be testing the prefix on every related resource type instead of just the individual resource type. If a resource was created with no other resources relating to it, it's idprefix would not be checked.

I believe using `resourceType` from line 289 just below it might be the proper place to do that validation.